rikai/experimental/pg/model.py

```diff
@@ -12,56 +11,77 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-from typing import Optional
+import json
+from typing import Any, Callable, Dict, Optional
 
 import torch
-
+import torchvision
+import torchvision.transforms as T
+from rikai.parquet.dataset import convert_tensor
+from rikai.pytorch.models.torch import ClassificationModelType, ObjectDetectionModelType
+from rikai.pytorch.models.feature_extractor import FeatureExtractorType
 from rikai.spark.sql.codegen.dummy import DummyModelSpec
 from rikai.spark.sql.codegen.fs import FileModelSpec
 from rikai.spark.sql.model import ModelType
 from rikai.types import Image
+from torchvision.models.feature_extraction import create_feature_extractor
+
+from .logging import info
+from .schema import parse_schema
 
 
 class PgModel:
+    """PostgreSQL Model"""
```

Comment:
nit: I think too many things are called "model" now. e.g.:
1. we the actual pytorch/tf "model"
2. we have the Rikai ModelType which gets assigned to `self.model`
3. And this PgModel is also a "model"

This is probably worth a separate discussion tho.

Reply:
Yea, thats true. This is more about a model adaptor